lib/ansible/modules/system/parted.py

```diff
@@ -0,0 +1,654 @@
+
+    return {
+        'generic': {
+            'dev':            device,
+            'table':          "unknown",
+            'size':           size,
+            'unit':           unit,
+            'logical_block':  logic_block,
+            'physical_block': phys_block,
+            'model':          "{0} {1}".format(vendor, model),
+        },
+        'partitions': []
+    }
+
+
+def get_device_info(device, unit):
+    """
+    Fetches information about a disk and its partitions and it returns a
+    dictionary.
+    """
+    global module
+
+    # If parted complains about missing labels, it means there are no partitions.
+    # In this case only, use a custom function to fetch information and emulate
+    # parted formats for the unit.
+    label_needed = check_parted_label(device)
+    if label_needed:
+        return get_unlabeled_device_info(device, unit)
+
+    command = "parted -s -m {0} -- unit '{1}' print".format(device, unit)
+    rc, out, err = module.run_command(command)
+    if rc != 0 and 'unrecognised disk label' not in err:
+        module.fail_json(msg=(
+            "Error while getting device information with parted "
+            "script: '{0}'".format(command)),
+            rc=rc, out=out, err=err
+        )
+
+    return parse_partition_info(out, unit)
+
+
+def check_parted_label(device):
+    """
+    Determines if parted needs a label to complete its duties. Versions prior
+    to 3.1 don't return data when there is no label. For more information see:
+    http://upstream.rosalinux.ru/changelogs/libparted/3.1/changelog.html
+    """
+    # Check the version
+    parted_major, parted_minor = parted_version()
+    if (parted_major == 3 and parted_minor >= 1) or parted_major > 3:
+        return False
+
+    # Older parted versions return a message in the stdout and RC > 0.
+    rc, out, err = module.run_command("parted -s -m {0} print".format(device))
+    if rc != 0 and 'unrecognised disk label' in out.lower():
+        return True
+
+    return False
+
+
+def parted_version():
+    """
+    Returns the major and minor version of parted installed on the system.
+    """
+    global module
+
+    rc, out, err = module.run_command("parted --version")
+    if rc != 0:
+        module.fail_json(
+            msg="Failed to get parted version.", rc=rc, out=out, err=err
+        )
+
+    lines = out.split('\n')
+    if len(lines) == 0:
+        module.fail_json(msg="Failed to get parted version.", rc=0, out=out)
+
+    matches = re.search(r'^parted.+(\d+)\.(\d+)$', lines[0])
+    if len(matches.groups()) != 2:
+        module.fail_json(msg="Failed to get parted version.", rc=0, out=out)
+
+    major = int(matches.group(1))
+    minor = int(matches.group(2))
+
+    return major, minor
+
+
+def parted(script, device, align):
+    """
+    Runs a parted script.
+    """
+    global module
+
+    if script and not module.check_mode:
+        command = "parted -s -m -a {0} {1} -- {2}".format(align, device, script)
```

Comment:
Using format() here means it won't work with Python 2.6 and older. Since Ansible supports Python 2.4 and higher you should really be using the old formatting method.
See http://pyformat.info/